src/tigerbeetle-java/src/main/java/com/tigerbeetle/Accounts.java

```diff
@@ -0,0 +1,317 @@
+     * @throws NullPointerException if {@code id} is null.
+     * @throws IllegalArgumentException if {@code id} is not 16 bytes long.
+     * @throws IllegalStateException if not at a {@link #isValidPosition valid position}.
+     * @throws IllegalStateException if a {@link #isReadOnly() read-only} batch.
+     */
+    public void setId(final byte[] id) {
+
+        if (id == null)
+            throw new NullPointerException("Id cannot be null");
+
+        putUInt128(at(Struct.Id), id);
+    }
+
+
+    /**
+     * Gets an optional secondary identifier to link this account to an external entity.
+     *
+     * @return an array of 16 bytes representing the 128-bit value.
+     * @throws IllegalStateException if not at a {@link #isValidPosition valid position}.
+     */
+    public byte[] getUserData() {
+        return getUInt128(at(Struct.UserData));
+    }
+
+    /**
+     * Gets an optional secondary identifier to link this account to an external entity.
+     *
+     * @param part a {@link UInt128} enum indicating which part of the 128-bit value is to be
+     *        retrieved.
+     * @return a {@code long} representing the the first 8 bytes of the 128-bit value if
+     *         {@link UInt128#LeastSignificant} is informed, or the last 8 bytes if
+     *         {@link UInt128#MostSignificant}.
+     * @throws IllegalStateException if not at a {@link #isValidPosition valid position}.
+     */
+    public long getUserData(final UInt128 part) {
+        return getUInt128(at(Struct.UserData), part);
+    }
+
+    /**
+     * Sets an optional secondary identifier to link this account to an external entity.
+     *
+     * @param leastSignificant a {@code long} representing the the first 8 bytes of the 128-bit
+     *        value.
+     * @param mostSignificant a {@code long} representing the the last 8 bytes of the 128-bit value.
+     * @throws IllegalStateException if not at a {@link #isValidPosition valid position}.
+     * @throws IllegalStateException if a {@link #isReadOnly() read-only} batch.
+     */
+    public void setUserData(final long leastSignificant, final long mostSignificant) {
+        putUInt128(at(Struct.UserData), leastSignificant, mostSignificant);
+    }
+
+    /**
+     * Sets an optional secondary identifier to link this account to an external entity.
+     * <p>
+     * May be zero, null values are converted to zero.
+     *
+     * @param userData an array of 16 bytes representing the 128-bit value.
+     * @throws IllegalArgumentException if {@code userData} is not 16 bytes long.
+     * @throws IllegalStateException if not at a {@link #isValidPosition valid position}.
+     * @throws IllegalStateException if a {@link #isReadOnly() read-only} batch.
+     */
+    public void setUserData(final byte[] userData) {
+        putUInt128(at(Struct.UserData), userData);
+    }
+
+    /**
+     * Gets an identifier that partitions the sets of accounts that can transact.
```

Comment:
This sentence is pretty confusing, and I see it was just taken from the real docs! I need to fix that.

Reply:
At the very least what do you think about having each getter/setter link to the generic docs like https://docs.tigerbeetle.com/reference/accounts#ledger ? Because you're only taking one or two sentences whereas the full docs have more info

Reply:
Great idea—the less we need to keep the comments in sync with the docs the better. Ideally, we just point directly there for everything.